### ISSUE TITLE
vscode-cpptools crashes when using constexpr and operator overloading

### ISSUE THREAD

**Issue by sewbacca** [microsoft/vscode-cpptools#9058]
Bug type: Language Service

**Describe the bug**
- OS and Version: Windows 10
- VS Code Version: 1.65.2
- C/C++ Extension Version: 1.8.4
- Other extensions you installed (and if the issue persists after disabling them): Using only C/C++
- If using SSH remote, specify OS of remote machine: No

When using the example code below the extension crashes (possibly because of the unwanted implicit recursive nature).

**Steps to reproduce**

1. Attach debugger to cpptools-srv.exe
2. Create a project with `main.cpp` and the code from below, so that intellisense is parsing it (using C++17 standard)
3. See the crash
When process is repeated without the `constexpr` or `~e` is replaced with `int(e)` , intellisense won't crash.

**Expected behavior**
No crash.

**Code sample**
- Code sample

```c++
#include <iostream>

enum MyEnum
{
    A = 0b01,
    B = 0b10,
    C = 0b11,
    D = 0b00,
};

constexpr MyEnum operator~(MyEnum e)
{
    return MyEnum(~e & 0b11); // For a correctly working program and intellisense replace `~e` with `~int(e)`
}

int main(int, char**)
{
    std::cout << ~A << std::endl;
}
```

**Comment by michelleangela** [comment]
Created an internal issue for IntelliSense parser that is shared between VS C++ and this VS Code C++ extension.

**Comment by michelleangela** [comment]
The internal issue is in a private repository and the IntelliSense is proprietary.

**Comment by sean-mcmanus** [comment]
@sewbacca The main reason we create a private issue is that the public issues on the developer community site get deleted every year or so due to their data retention policies. We use this public issue to track it (although we could post the internal bug number).

**Comment by sean-mcmanus** [comment]
This should be fixed with 1.10.2 (pre-release), although it was fixed in 1.10.0 as well.
